lib/RequestRouter.js

```diff
@@ -463,7 +436,7 @@ RequestRouter.prototype.setResource = function(resource) {
 
     var schema = this._schema;
 
-    if (this._request.getMethod() === Request.METHOD_POST && this.isCollection()) {
+    if (this.isCollection()) {
```

Comment:
This doesn't look right. We need to use the collection schema unless it's a POST, then we use the item schema.